lib/CPathfinder.h

```diff
@@ -121,4 +183,36 @@ class CPathfinder : private CGameInfoCallback
 	bool addTeleportOneWay(const CGTeleport * obj) const;
 	bool addTeleportOneWayRandom(const CGTeleport * obj) const;
 	bool addTeleportWhirlpool(const CGWhirlpool * obj) const;
+
+	bool canVisitObject() const;
+
+};
+
+struct TurnInfo
+{
+	int turn;
+	int maxMovePointsLand;
+	int maxMovePointsWater;
+	const Bonus * bonusFlying;
+	const Bonus * bonusWaterWalking;
+};
+
+class DLL_LINKAGE CPathfinderHelper
+{
+public:
+	TurnInfo * ti;
+	const CGHeroInstance * hero;
+
+	CPathfinderHelper(const CGHeroInstance * Hero);
+	void updateTurnInfo(const int &turn = 0);
```

Comment:
Use const value but not reference for basic types like int. (But reference is recommended for structures)